dep: `libwebp-tools`

130s opened the pull request:
<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

## Package name:

`webp`

## Package Upstream Source:

Likely [official page (developers.google.com)](https://developers.google.com/speed/webp)

Note this is different from libwebp, of which dep key has already been added https://github.com/ros/rosdistro/pull/12531.

## Purpose of using this:

Newer alternative for image file format.

Mentioned also in https://github.com/ros/rosdistro/pull/14163#issuecomment-286455819 by former perception pipeline maintainer who also seems to be one of webp developers https://github.com/ros/rosdistro/pull/12531.

## Links to Distribution Packages

<!-- Replace the REQUIRED areas with the URL to the package.  For IF AVAILABLE areas, either put in the URL to the package or state 'not available'.
More info at https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules -->

- Debian: https://packages.debian.org/search?keywords=webp
- Ubuntu: https://packages.ubuntu.com/search?keywords=webp
- Fedora: https://packages.fedoraproject.org/pkgs/libwebp/libwebp-tools/
- :x: Arch: https://archlinux.org/packages/extra/x86_64/libwebp-tools/ -> 404
   - The description of `libwebp` pkg reads as follows, i.e. includes "tools". But looking at the list of files https://archlinux.org/packages/extra/x86_64/libwebp/files/, I do not see e.g. `vwebp`, the very tool I like to use. So I'd say no.
      > WebP library and conversion tools
- macOS: https://formulae.brew.sh/formula/webp
- Alpine: https://pkgs.alpinelinux.org/package/edge/main/x86_64/libwebp-tools
- openSUSE: https://software.opensuse.org/package/libwebp-tools

130s committed:
dep: webp

Signed-off-by: Isaac Saito <iisaac.saito@gmail.com>

cottsay commented:
It looks like the tools provided by the `webp` package on Ubuntu are part of the `webp-tools` package for Fedora: https://packages.fedoraproject.org/pkgs/libwebp/libwebp-tools/

130s committed:
dep: libwebp-tools: Renamed from 'webp' per suggestion https://github.com/ros/rosdistro/pull/43326#issuecomment-2452306585

Signed-off-by: Isaac Saito <iisaac.saito@gmail.com>

130s commented:
@cottsay Thanks for the finding. I updated the PR with more distros. Indeed "`libwebp-tools`" seems commonly used as well. In my limited finding (posted in OP) both "`webp`" and "`libwebp-tools`" are used.

For this particular case, I feel including "tools" seems more reasonable so I renamed the rosdep key to "`libwebp-tools`", but I don't know. I'm ok with either naming.

I'm not sure if there are other cases where the pkg name used in Debian/Ubuntu isn't used as a rosdep key, given Ubuntu might be the most used OS among `rosdep` users.

130s committed:
dep: libwebp-tools: Fix rhel

Co-authored-by: Scott K Logan <logans@cottsay.net>

wjwwood reviewed (approved):
lgtm, @cottsay may want to review again though.
